$NetBSD: patch-src_meson.build,v 1.1 2025/07/01 11:35:06 adam Exp $

Compiler option -export-symbols is not portable; doesn't work with Clang.

--- src/meson.build.orig	2024-04-18 18:05:53.856125641 +0000
+++ src/meson.build
@@ -185,10 +185,15 @@ spice_client_glib_syms = files('map-file
 spice_client_glib_syms_path = meson.current_source_dir() / 'map-file'
 spice_gtk_version_script = '-Wl,--version-script=@0@'.format(spice_client_glib_syms_path)
 spice_gtk_has_version_script = compiler.has_link_argument(spice_gtk_version_script)
+spice_compiler_has_export_symbols = compiler.has_link_argument('-export-symbols')
 if not spice_gtk_has_version_script
   spice_client_glib_syms = files('spice-glib-sym-file')
   spice_client_glib_syms_path = meson.current_source_dir() / 'spice-glib-sym-file'
+  if spice_compiler_has_export_symbols
   spice_gtk_version_script = ['-export-symbols', spice_client_glib_syms_path]
+  else
+  spice_gtk_version_script = []
+  endif
 endif
 
 # soversion
@@ -376,7 +381,11 @@ if spice_gtk_has_gtk
   if not spice_gtk_has_version_script
     spice_client_gtk_syms = files('spice-gtk-sym-file')
     spice_client_gtk_syms_path = meson.current_source_dir() / 'spice-gtk-sym-file'
+    if spice_compiler_has_export_symbols
     spice_gtk_version_script = ['-export-symbols', spice_client_gtk_syms_path]
+    else
+    spice_gtk_version_script = []
+    endif
   endif
 
   # soversion
